### PR TITLE
chore(broker-core): listen to leaderelection primitive disconnect

### LINKED ISSUE #2325: Old leader doesnot detect leader change during network partition
With atomix leader election primitive, leadership event listeners get events only when they are connected to the quorum. When a leader change occurs due to a network partition between old leader and the other brokers, old leader (stream processor) does not step down.

### PR DESCRIPTION
**Pull request by deepthidevaki**
* a node can listen to network disconnection so that it can step down from stream processor leader
* there is now a PartitionLeaderElection service which all services can use to listen to leadership changes.

closes #2325

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]
Looks good. Just some small notes.

BTW: Can we now enable the test case `BrokerLeaderChangeTest#shouldChangeLeaderAfterLeaderDies()`?

**Comment by deepthidevaki** [comment]
> BTW: Can we now enable the test case `BrokerLeaderChangeTest#shouldChangeLeaderAfterLeaderDies()`?

I guess yes.

**Comment by deepthidevaki** [comment]
@saig0 I have made the suggested changes. Please have a look.

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/PartitionLeaderElection.java [hunk 1]
```diff
@@ -68,23 +92,150 @@ public void start(ServiceStartContext startContext) {
...
+    try {
+      final Leader<String> currentLeader = election.getLeadership().leader();
+      if (memberId.equals(currentLeader.id())) {
+        transitionToLeader(currentLeader.term());
+      } else {
+        transitionToFollower();
+      }
+    } catch (PrimitiveException e) {
+      LOG.error(
+          "Couldn't get current leadership for partition {}. Transitioning to follower. {}",
```
Comment (saig0) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/PartitionLeaderElection.java]:
```suggestion
          "Couldn't get current leadership for partition {}. Transitioning to follower.",
```

The exception doesn't need a parameter in the log statement. Just add it as the last argument.

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/PartitionLeaderElection.java [hunk 2]
```diff
@@ -68,23 +92,150 @@ public void start(ServiceStartContext startContext) {
...
+                transitionToFollower();
+              }
+            });
+        break;
+      case CONNECTED:
+        try {
+          final Leader<String> currentLeader = election.getLeadership().leader();
+          onLeadershipChange(currentLeader);
+        } catch (PrimitiveException e) {
+          LOG.error("Couldn't get current leadership for partition {} {}", partitionId, e);
```
Comment (saig0) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/PartitionLeaderElection.java]:
```suggestion
          LOG.error("Couldn't get current leadership for partition {}", partitionId, e);
```

**Review by saig0 — CHANGES_REQUESTED** [review]

**Comment by deepthidevaki** [comment]
@saig0 updated.

**Review by saig0 — APPROVED** [review]

**Pull request merged**